flytekit/annotated/workflow.py

```diff
@@ -289,11 +291,57 @@ def get_registerable_entity(self) -> _SdkWorkflow:
         return self._registerable_entity
 
 
-def workflow(_workflow_function=None):
+class ReferenceWorkflow(Workflow):
```

Comment:
should we make it such that a ReferenceWorkflow can be created inline, without needing a function. Something like this
```python
w = Workflow(reference=WorkflowReference(...), inputs=kwtypes(), outputs=kwtypes())
```

Reply:
I know you made the task also, to use an empty function, but if we factor it so, then we can implement both methods?

Reply:
How about just a `ReferenceEntity` but keep the other ones as is?